core_api/src/runnables.py

```diff
@@ -180,11 +179,30 @@ def make_condense_rag_runnable(
         RunnablePassthrough()
         | {
             "question": condense_question_chain,
-            "documents": retriever | format_chunks,
+            "documents": retriever | format_documents,
             "sources": retriever,
         }
         | {
             "response": prompt | llm | StrOutputParser(),
             "sources": itemgetter("sources"),
         }
     )
+
+
+def resize_documents(max_tokens: int | None = None) -> list[Document]:
```

Comment:
we're 100% reinventing the wheel here and I want to think about a new ingest pipeline which creates these larger chunks. I think langchain can help here with the ParentDocumentRetriever as you say so that'd be good to look at. We can remove so much faffing around with docs if we get that working